### PR TITLE
Downgrade self-hosted runner to cuda 12.1

### PR DESCRIPTION
**Pull request by ckolbPTB**
In order to run the cuda tests on our hpc we need very specific cuda versions and therefore pytorch versions. It is quite complicated to get this to work with our `--upgrade --upgrade-strategy eager` approach. 

I think it should be fine, if the self-hosted runner does not use the upgrade because any latest package changes will be tested with the cpu tests anyway. 

There are a few corner cases where the current approach for the cuda tests could fail but they should be caught by the `pip check` part.

### CODE REVIEW

File: docker/Dockerfile_runner [hunk 1]
```diff
@@ -27,21 +27,30 @@ RUN chmod -R 777 /actions-runner
...
 # install mrpro
 # forces rebuild on any change in the mrpro directory
 COPY . /mrpro/
-RUN python -m pip install "/mrpro[notebooks]" --no-cache-dir --upgrade --upgrade-strategy "eager" \
-    torch --extra-index-url https://download.pytorch.org/whl/${CUDA_VERSION}
-RUN rm -rf /mrpro
+RUN python -m pip install  "/mrpro[notebooks]" --no-cache-dir && rm -rf /mrpro
+
+# check for mrpro package errors
+RUN python -m pip check | tee /tmp/pip_check.txt; \
```
Comment (fzimmermann89) [inline docker/Dockerfile_runner:45]: This is to check if we installed a version that is lower than the minimum version requiered by mrpro, correct?

File: docker/Dockerfile_runner [hunk 2]
```diff
@@ -27,21 +27,30 @@ RUN chmod -R 777 /actions-runner
...
+# install mrpro
 WORKDIR /
 
+# install mrpro dependencies
+# forces rebuild if either the version or the pyproject.toml changes
 COPY docker/install_dependencies.sh pyproject.toml src/mrpro/VERSION ./
-
-# the 1st argument for the install_dependencies specifies the cuda version 12.4 for torch
+# the 1st argument for the install_dependencies specifies the cuda version
 RUN bash install_dependencies.sh ${CUDA_VERSION} && rm install_dependencies.sh pyproject.toml VERSION
```
Comment (fzimmermann89) [inline docker/Dockerfile_runner]: this provides CUDA_VERSION as the first argument, but this is no longer used in install_depencencies?

**Review by fzimmermann89 — APPROVED** [review]

File: docker/Dockerfile_runner (hunk 1 above)
Comment (ckolbPTB) [reply docker/Dockerfile_runner:45]: yes, exactly
